script/setup.js

```diff
@@ -197,191 +234,296 @@ try {
-				required_status_checks: {
-					checks: [
-						{ context: "build" },
-						{ context: "compliance" },
-						{ context: "knip" },
-						{ context: "lint" },
-						{ context: "markdown" },
-						{ context: "package" },
-						{ context: "packages" },
-						{ context: "prettier" },
-						{ context: "spelling" },
-						{ context: "test" },
-					],
-					strict: false,
-				},
-				restrictions: null,
-			}
-		);
+		s.stop(chalk.green`✅ Hydrated branch protection settings.`);
 
-		console.log(chalk.gray`✔️ Done.`);
+		s.start(`Finishing API hydration...`);
+		s.stop(chalk.green`✅ Finished API hydration.`);
 	}
 
-	console.log();
-	console.log(chalk.gray`Removing setup script...`);
+	s.start(`Removing setup script...`);
 
-	await fs.rm("./script", { force: true, recursive: true });
-	await fs.rm(".github/workflows/setup.yml");
+	try {
+		await rm("./script", { force: true, recursive: true });
+		await rm(".github/workflows/setup.yml");
+	} catch (error) {
+		s.stop(chalk.red`❌ Error removing setup script.`);
 
-	console.log(chalk.gray`✔️ Done.`);
-} catch (error) {
-	console.log(chalk.red(error.stack));
-	caughtError = error;
-} finally {
-	console.log();
-	console.log(
-		chalk.gray`Removing devDependency packages only used for setup...`
-	);
+		throw new Error("Could not remove setup script. ", {
+			cause: error,
+		});
+	}
+
+	s.stop(chalk.green`✅ Removed setup script.`);
+
+	s.start(`Removing devDependency packages only used for setup...`);
 
 	try {
-		await $`pnpm remove @clack/prompts all-contributors-cli chalk octokit npm-user replace-in-file title-case -D`;
+		await $`pnpm remove @clack/prompts all-contributors-cli chalk octokit npm-user replace-in-file titleize read-pkg-up load-json-file git-remote-origin-url git-url-parse execa -D`;
 	} catch (error) {
-		console.log("Error uninstalling packages:", error);
-		caughtError = error;
+		s.stop(
+			chalk.red`❌ Error removing devDependency packages only used for setup`
+		);
+
+		throw new Error("Could not uninstalling packages. ", {
+			cause: error,
+		});
 	}
 
-	console.log(chalk.gray`✔️ Done.`);
-}
+	s.stop(chalk.green`✅ Removed devDependency packages only used for setup. `);
 
-console.log();
+	outro(chalk.green`Great, looks like everything worked! 🎉`);
 
-if (caughtError) {
-	console.log(
-		chalk.yellow`Looks like there was a problem. Correct it and try again? 😕`
-	);
-} else {
-	console.log(chalk.green`Great, looks like everything worked!`);
 	console.log(chalk.blue`You may consider committing these changes:`);
-	console.log(chalk.gray`\tgit add -A`);
-	console.log(chalk.gray`\tgit commit -m "chore: hydrated repo"`);
-	console.log(chalk.gray`\tgit push`);
+	console.log();
+	console.log(chalk.gray`git add -A`);
+	console.log(chalk.gray`git commit -m "chore: hydrated repo"`);
+	console.log(chalk.gray`git push`);
+	console.log();
 	console.log(chalk.greenBright`See ya! 👋`);
-}
+	console.log();
+
+	exitCode = 0;
+} catch (error) {
+	s.start(`Restoring the changes done... `);
```

Comment:
This feature of using `git` to restore changes is a great idea that should also be a separate PR. I'd want it to prompt the user to ask if they want to do that. Sometimes it mostly works & that alone is enough to setup (or debug why it failed).

Reply:
Right, we can move this to separate PR & ask this the prompt.